tasking/pulp/tasking/worker_watcher.py

```diff
@@ -72,14 +72,16 @@ def handle_worker_heartbeat(event):
     :type event: dict
     """
     event_info = _parse_and_log_event(event)
-    worker = Worker.objects(name=event_info['worker_name']).first()
 
-    if not worker:
+    try:
+        worker = Worker.objects.get(name=event_info['worker_name'])
+    except worker.DoesNotExist:
+        worker = Worker(name=event_info['worker_name'])
```

Comment:
I think this would be better using [`worker, created = Worker.objects.get_or_create(name=event_info['worker_name'])`](https://docs.djangoproject.com/en/1.10/ref/models/querysets/#get-or-create)